azure-toolkit-libs/azure-toolkit-springcloud-lib/src/main/java/com/microsoft/azure/toolkit/lib/springcloud/task/DeploySpringCloudAppTask.java

```diff
@@ -69,6 +70,7 @@ public DeploySpringCloudAppTask(SpringCloudAppConfig appConfig) {
         final boolean toCreateApp = !app.exists();
         final boolean toCreateDeployment = !deployment.exists();
 
+        AzureTelemetry.getContext().getActionProperties().put("subscriptionId", config.getSubscriptionId());
```

Comment:
How about track subscription id while get azure client in case subscription is not set in `pom.xml` using the default value.